### PR TITLE
refactor(infra): refactor copilot client

### PR DESCRIPTION
**Pull request by EYHN**

### CODE REVIEW

**Comment by EYHN** [comment]
* **#8813** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8813?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8813?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#8745** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8745?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by forehalo — APPROVED** [review]

File: packages/frontend/core/src/components/providers/workspace-side-effects.tsx [hunk 1]
```diff
@@ -129,6 +137,20 @@ export const WorkspaceSideEffects = () => {
     };
   }, [globalDialogService]);
 
+  const graphqlService = useService(GraphQLService);
+  const eventSourceService = useService(EventSourceService);
+  const fetchService = useService(FetchService);
+
+  useEffect(() => {
+    setupAIProvider(
```
Comment (pengx17) [inline packages/frontend/core/src/components/providers/workspace-side-effects.tsx]: seems `setupAIProvider` has some subscriptions that were not unsubscribed which were ok before. We may need to clean them up if having configurable servers.

**Review by pengx17 — APPROVED** [review]

File: packages/frontend/core/src/components/providers/workspace-side-effects.tsx (hunk 1 above)
Comment (EYHN) [reply packages/frontend/core/src/components/providers/workspace-side-effects.tsx]: It seems that each time setupAIProvider() will overwrite the previous one?

File: packages/frontend/core/src/components/blocksuite/block-suite-editor/ai/setup-provider.tsx [hunk 2]
```diff
@@ -460,7 +492,7 @@ Could you make a new website based on these notes and send back just the html fi
   AIProvider.provide('onboarding', toggleGeneralAIOnboarding);
 
   AIProvider.provide('forkChat', options => {
-    return forkCopilotSession(options);
+    return forkCopilotSession(client, options);
   });
 
   AIProvider.slots.requestLogin.on(() => {
```
Comment (pengx17) [inline packages/frontend/core/src/components/blocksuite/block-suite-editor/ai/setup-provider.tsx]: ⬆️here

**Pull request merged**